app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt

```diff
@@ -25,20 +26,40 @@ class ParkingViewModel(
   private val _rectParkings = MutableStateFlow<List<Parking>>(emptyList())
   val rectParkings: StateFlow<List<Parking>> = _rectParkings
 
-  /** List of k (or less) closest parking queried */
-  private val _kClosestParkings = MutableStateFlow<List<Parking>>(emptyList())
-  val kClosestParkings: StateFlow<List<Parking>> = _kClosestParkings
+  /** List of parkings in the circle of radius _radius around _circleCenter */
+  private val _closestParkings = MutableStateFlow<List<Parking>>(emptyList())
+  val closestParkings: StateFlow<List<Parking>> = _closestParkings
 
   /** Selected parking to review/edit */
   private val _selectedParking = MutableStateFlow<Parking?>(null)
   val selectedParking: StateFlow<Parking?> = _selectedParking
 
+  // Circle center and radius for the circle search for the list screen
+  private val _radius = MutableStateFlow(0.0)
+  private val _circleCenter = MutableStateFlow<Point?>(null)
   // List of tiles to display
   private var tilesToDisplay: Set<Tile> = emptySet()
   // Map a tile to the parkings that are in it.
   private val tilesToParking =
       MutableStateFlow<LinkedHashMap<Tile, List<Parking>>>(LinkedHashMap(10, 1f, true))
 
+  init {
+    viewModelScope.launch {
```

Comment:
This is the core of the feature of the PR. 
The _rectParkings state is now linked to the _closestParkings state.

In a way  _closestParkings is just  _rectParkings sorted and filtered.